physics/cires_ugwp_v1.F90

```diff
@@ -0,0 +1,635 @@
+!>  \file cires_ugwp_v1.F90
+!! This file contains the Version 1 Unified Gravity Wave Physics (UGWP) scheme by Valery Yudin (University of Colorado, CIRES)
+!! See Valery Yudin's presentation at 2017 NGGPS PI meeting:
+!! Gravity waves (GWs): Mesoscale GWs transport momentum, energy (heat), and create eddy mixing in the whole atmosphere domain; Breaking and
+!dissipating GWs deposit: (a) momentum; (b) heat (energy); and create (c) turbulent mixing of momentum, heat, and tracers
+!! To properly incorporate GW effects (a-c) unresolved by DYCOREs we need GW physics
+!! "Unified": a) all GW effects due to both dissipation/breaking; b) identical GW solvers for all GW sources; c) ability to replace solvers.
+!! Unified Formalism:
+!! 1. GW Sources: Stochastic and physics based mechanisms for GW-excitations in the lower atmosphere, calibrated by the high-res analyses/forecasts, and observations (3 types of GW sources: orography, convection, fronts/jets).
+!! 2. GW Propagation: Unified solver for "propagation, dissipation and breaking" excited from all type of GW sources.
+!! 3. GW Effects: Unified representation of GW impacts on the "resolved" flow for all sources (energy-balanced schemes for momentum, heat and mixing).
+!!
+!! https://www.weather.gov/media/sti/nggps/Presentations%202017/02%20NGGPS_VYUDIN_2017_.pdf
+
+module cires_ugwp_v1
+
+    use machine, only: kind_phys
+
+    use cires_ugwp_module_v1, only: knob_ugwp_version, cires_ugwp_init_v1, &
+                                 cires_ugwp_finalize_v1
+
+    use gwdps, only: gwdps_run
+
+    implicit none
+
+    private
+
+    public cires_ugwp_v1_run
+
+    logical :: is_initialized = .False.
+
+
+contains
+
+
+
+! -----------------------------------------------------------------------
+! CCPP entry points for CIRES Unified Gravity Wave Physics (UGWP) scheme v0
+! -----------------------------------------------------------------------
+!    originally from cires_ugwp_driver_v1.f
+!    driver of cires_ugwp   (_driver)
+! -----------------------------------------------------------------------
+!   driver is called after pbl & before chem-parameterizations
+! -----------------------------------------------------------------------
+!  order = dry-adj=>conv=mp-aero=>radiation -sfc/land- chem -> vertdiff-> [rf-gws]=> ion-re
+! -----------------------------------------------------------------------
+!>@brief The subroutine initializes the CIRES UGWP
+!> \section arg_table_cires_ugwp_init Argument Table
+!! \htmlinclude cires_ugwp_v1_init.html
+!!
+! -----------------------------------------------------------------------
+!
+
+
+subroutine cires_ugwp_v1_init(Model,Init_parm,Grid)
```

Comment:
Please do not pass in the GFS DDTs - pass in the individual components that you need (Model%me, ...) - we do not want dependency on the GFS DDTs.